### PR TITLE
Logs: rolling files

### PR DESCRIPTION
**Pull request by Mathieu-Lala**
# App behavior changes : 

The logs (server and app) are written in `logs.filepath` until reaching `logs.size_limit`.
After that the file is truncated and the previous logs are archived into `{logs.filepath}-ar/trace.{count}.gz` ; until `archive_count`.

# Details of implementation changes :

* remove unused features of `log4rs`
* use rolling file instead of flat files (unlimited size)
* add `size_limit` parameter for `server.logs` and `app.logs`
* add `archive_count` parameter for `server.logs` and `app.logs`
* maintain the tests

# Improvement

* [ ] `size_limit` is an `u64` and could be deserialized from a string like so : `"20GB"`, `"1.5KB"`, ...

### CODE REVIEW

File: vsmtp-config/src/log4rs_helper.rs [hunk 1]
```diff
@@ -1,24 +1,58 @@
...
+            Box::new(CompoundPolicy::new(
+                Box::new(trigger::size::SizeTrigger::new(size_limit)),
+                Box::new(
+                    roll::fixed_window::FixedWindowRoller::builder()
+                        .base(0)
+                        .build(
+                            &format!("{}-ar/trace.{{}}.gz", path.display()),
+                            archive_count,
+                        )
+                        .unwrap(),
```
Comment (Mathieu-Lala) [inline vsmtp-config/src/log4rs_helper.rs:33]: Raise Result::Error

File: vsmtp-config/src/log4rs_helper.rs [hunk 2]
```diff
@@ -1,24 +1,58 @@
...
-        .encoder(Box::new(encode::pattern::PatternEncoder::new(
-            &config.app.logs.format,
-        )))
-        .build(&config.app.logs.filepath)
-        .with_context(|| format!("For filepath: '{}'", config.app.logs.filepath.display()))?;
+    let server = init_rolling_log(
+        &config.server.logs.format,
+        &config.server.logs.filepath,
+        config.app.logs.size_limit,
+        config.app.logs.archive_count,
```
Comment (Mathieu-Lala) [inline vsmtp-config/src/log4rs_helper.rs:48]: `config.server`

File: vsmtp-config/src/log4rs_helper.rs [hunk 3]
```diff
@@ -1,24 +1,57 @@
 use crate::{log_channel, Config};
 use vsmtp_common::re::{anyhow, log};
 
+fn init_rolling_log(
+    format: &str,
+    filepath: &std::path::Path,
+    size_limit: u64,
+    archive_count: u32,
+) -> anyhow::Result<log4rs::append::rolling_file::RollingFileAppender> {
+    use anyhow::Context;
```
Comment (ltabis) [inline vsmtp-config/src/log4rs_helper.rs:10]: Why did you choose to put the use statement here ?

**Review by ltabis — DISMISSED** [review]

File: vsmtp-config/src/log4rs_helper.rs (hunk 3 above)
Comment (Mathieu-Lala) [reply vsmtp-config/src/log4rs_helper.rs:10]: Yes could be at file top level

**Review by ltabis — APPROVED** [review]

**Pull request merged**